crt_portal/cts_forms/views.py

```diff
@@ -635,10 +640,52 @@ def post(self, request):
             return render(request, 'forms/complaint_view/actions/index.html', output)
 
 
-class SaveReportAttachmentView(LoginRequiredMixin, FormView):
+class ReportAttachmentView(LoginRequiredMixin, FormView):
     """Can be used for saving attachments for a report"""
     form_class = AttachmentActions
-    http_method_names = ['post']
+    http_method_names = ['get', 'post']
+
+    def get(self, request, id, filename):
+        """
+        Download a particular attachment for a report
+        """
+        report = get_object_or_404(Report.objects.prefetch_related('attachments'), pk=id)
+        attachment = report.attachments.get(file=f'attachments/{filename}')
+
+        logger.info(f'User {request.user} downloading attachment {attachment.filename} ({attachment.file.name}) for report {id}')
```

Comment:
I got caught up a bit here untangling the difference between, `filename`, `attachment.filename`, and `attachment.file.name`. 

Is there a way we can disambiguate these now before they become an acceptable pattern for us in the future?

Using `filename` in the URL doesn't gain us much. Accessing these as `attachments/{attachment.id}` could be a bit simpler to handle.

Reply:
simplified as per suggestions.

Reply:
Since we have `attachment_id` in the url now we should get this directly by attachment ID

Reply:
hm, I'm seeing ☝️ this `attachment_id` change - GitHub might be showing an outdated version @Jkrzy, which happens to me all too frequently

Reply:
:woah: You're right, my mistake. Thanks!